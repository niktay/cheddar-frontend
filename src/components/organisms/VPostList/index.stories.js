import { storiesOf } from "@storybook/vue";
import VPostList from ".";

storiesOf("Organism - VPostList", module).add("default", () => ({
  components: { VPostList },

  data() {
    return {
      posts: [
        {
          title: "First Sample Title",
          author: {
            firstName: "John",
            lastName: "Doe"
          },
          content: `Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptate mollitia quaerat voluptatem 
                    optio obcaecati quos ex saepe. Minus, optio quis! Labore fugit deleniti ex id commodi ab nobis eaque 
                    adipisci?`
        },

        {
          title: "Second Sample Title",
          author: {
            firstName: "Jane",
            lastName: "Doe"
          },
          content: `Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptate mollitia quaerat voluptatem 
                    optio obcaecati quos ex saepe. Minus, optio quis! Labore fugit deleniti ex id commodi ab nobis eaque 
                    adipisci?`
        }
      ]
    };
  },

  template: `<VPostList :posts="posts"/>`
}));
