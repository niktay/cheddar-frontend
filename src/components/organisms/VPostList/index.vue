<template>
  <div class="o-VPostList">
    <VPostItem v-for="post in posts" :key="post.id" :post="post" />
  </div>
</template>

<script>
import VPostItem from "@/components/molecules/VPostItem/.";

export default {
  components: {
    VPostItem
  },

  props: {
    posts: {
      type: Array,
      required: true
    }
  }
};
</script>

<style scoped lang="scss">
.o-VPostList {
  grid-column: 2/3;
}
</style>
