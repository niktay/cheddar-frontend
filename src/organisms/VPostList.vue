<template>
  <div class="o-VPostList">
    <VPostItem v-for="post in posts" :key="post.id" :post="post" />
  </div>
</template>

<script>
import { ALL_POSTS_QUERY } from "../constants/graphql";
import VPostItem from "@/molecules/VPostItem";

export default {
  components: {
    VPostItem
  },

  props: {
    posts: {
      type: Array,
      required: true
    }
  }
};
</script>

<style scoped lang="scss">
.o-VPostList {
  grid-column: 2/3;
}
</style>
