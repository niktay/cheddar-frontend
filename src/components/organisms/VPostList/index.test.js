import { mount } from "@vue/test-utils";
import VPostList from ".";

describe("VPostItem component", () => {
  it("should render correctly", () => {
    const comp = mount(VPostList, {
      propsData: {
        posts: [
          {
            title: "First Sample Title",
            author: {
              firstName: "John",
              lastName: "Doe"
            },
            content: `Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptate mollitia quaerat voluptatem 
                      optio obcaecati quos ex saepe. Minus, optio quis! Labore fugit deleniti ex id commodi ab nobis eaque 
                      adipisci?`
          },

          {
            title: "Second Sample Title",
            author: {
              firstName: "Jane",
              lastName: "Doe"
            },
            content: `Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptate mollitia quaerat voluptatem 
                      optio obcaecati quos ex saepe. Minus, optio quis! Labore fugit deleniti ex id commodi ab nobis eaque 
                      adipisci?`
          }
        ]
      }
    });

    expect(comp.html()).toMatchSnapshot();
  });
});
