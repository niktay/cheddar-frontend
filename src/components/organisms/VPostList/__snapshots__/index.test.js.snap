// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VPostItem component should render correctly 1`] = `
<div class="o-VPostList">
  <article class="m-VPostItem">
    <h2 class="a-VPostTitle">First Sample Title</h2> <span class="a-VPostAuthor">
  by John Doe
</span>
    <p class="a-VPostContent">Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptate mollitia quaerat voluptatem
      optio obcaecati quos ex saepe. Minus, optio quis! Labore fugit deleniti ex id commodi ab nobis eaque
      adipisci?</p>
  </article>
  <article class="m-VPostItem">
    <h2 class="a-VPostTitle">Second Sample Title</h2> <span class="a-VPostAuthor">
  by Jane Doe
</span>
    <p class="a-VPostContent">Lorem ipsum dolor, sit amet consectetur adipisicing elit. Voluptate mollitia quaerat voluptatem
      optio obcaecati quos ex saepe. Minus, optio quis! Labore fugit deleniti ex id commodi ab nobis eaque
      adipisci?</p>
  </article>
</div>
`;
